### ISSUE TITLE
[BUG] anvi-run-hmms gives an unequal number of fields error in snakemake workflow

### ISSUE THREAD

**Issue by efogarty11** [merenlab/anvio#1530]
## Short description of the problem

`anvi-run-hmms` gives an unequal number of fields error when running in a snakemake workflow.  

## anvi'o version

anvi'o main on midway. 

## System info

Please tell us which operating system you are using, and how did you install anvi'o.

## Detailed description of the issue

When I run `anvi-run-hmms` from a snakemake workflow I get this output:

```
Target found .................................: RNA:CONTIG
Target found .................................: AA:GENE

HMM Profiling for Ribosomal_RNAs
===============================================
Reference ....................................: Seemann T,
                                                https://github.com/tseemann/barrnap
Kind .........................................: Ribosomal_RNAs
Alphabet .....................................: RNA
Context ......................................: CONTIG
Domain .......................................: N/A
HMM model path ...............................: /tmp/jobs/6276737/tmp7v39rkmb/Ribosomal_RNAs.hmm
Number of genes in HMM model .................: 12
Noise cutoff term(s) .........................: --cut_ga
Number of CPUs will be used for search .......: 8
HMMer program used for search ................: nhmmscan
Temporary work dir ...........................: /tmp/jobs/6276737/tmpiapa_ln8
Log file for thread 0 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.0_log
Log file for thread 1 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.1_log
Log file for thread 2 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.2_log
Log file for thread 3 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.3_log
Log file for thread 4 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.4_log
Log file for thread 5 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.5_log
Log file for thread 6 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.6_log
Log file for thread 7 ........................: /tmp/jobs/6276737/tmpiapa_ln8/RNA_contig_sequences.fa.7_log
Number of raw hits ...........................: 21

✖ anvi-run-hmms encountered an error after 0:00:17.517374


File/Path Error: Not all lines in the file '/tmp/tmpux6afx_6/hmm.table' have equal
                 number of fields...
```

## Files to reproduce

**The contigs.db that I was running this on is on slack under the notification for this issue.** 

Here is the `/tmp/tmpux6afx_6/hmm.table' file that is produced:
```
Bacterial_23S_rRNA	-	K_intermedius_KB2_B1_000000000078	-	11	2901	3026	300	3036	295	2906	-	0	2385.5	61.
Bacterial_16S_rRNA	RF00177	K_intermedius_KB2_B1_000000000078	-	5	1533	5251	3767	5255	3767	1533	-	0	1351.1	37.
Archaeal_16S_rRNA	RF01959	K_intermedius_KB2_B1_000000000078	-	7	1477	5244	3769	5249	3769	1477	-	1.5e-231	763.6	36.
Bacterial_23S_rRNA	-	K_rhaeticus_CBB_B2_000000000048	-	11	2900	3053	328	3063	322	2906	-	0	2384.1	63.6
Bacterial_16S_rRNA	RF00177	K_rhaeticus_CBB_B2_000000000048	-	5	1533	5304	3820	5308	3820	1533	-	0	1353.7	36.3
Archaeal_16S_rRNA	RF01959	K_rhaeticus_CBB_B2_000000000048	-	7	1477	5297	3822	5302	3822	1477	-	8.6e-232	764.4	35.7
Bacterial_23S_rRNA	-	K_rhaeticus_KBJ1_B3_000000000052	-	11	2900	3056	331	3066	325	2906	-	0	2384.1	63.6
Bacterial_16S_rRNA	RF00177	K_rhaeticus_KBJ1_B3_000000000052	-	5	1533	5307	3823	5311	3823	1533	-	0	1353.7	36.3
Archaeal_16S_rRNA	RF01959	K_rhaeticus_KBJ1_B3_000000000052	-	7	1477	5300	3825	5305	3825	1477	-	8.6e-232	764.4	35.7
Bacterial_23S_rRNA	-	K_rhaeticus_M2_B4_000000000086	-	11	2900	3056	331	3066	325	2906	-	0	2384.1	63.6
Bacterial_16S_rRNA	RF00177	K_rhaeticus_M2_B4_000000000086	-	5	1533	5307	3823	5311	3823	1533	-	0	1353.7	36.3
Archaeal_16S_rRNA	RF01959	K_rhaeticus_M2_B4_000000000086	-	7	1477	5300	3825	5305	3825	1477	-	8.6e-232	764.4	35.7
Bacterial_23S_rRNA	-	K_rhaeticus_MF_B2_000000000084	-	11	2900	3056	331	3066	325	2906	-	0	2384.1	63.6
Bacterial_16S_rRNA	RF00177	K_rhaeticus_MF_B2_000000000084	-	5	1533	5307	3823	5311	3823	1533	-	0	1353.7	36.3
Archaeal_16S_rRNA	RF01959	K_rhaeticus_MF_B2_000000000084	-	7	1477	5300	3825	5305	3825	1477	-	8.6e-232	764.4	35.7
Bacterial_23S_rRNA	-	K_rhaeticus_N_B2_000000000041	-	11	2900	3056	331	3066	325	2906	-	0	2384.1	63.6	-
Bacterial_16S_rRNA	RF00177	K_rhaeticus_N_B2_000000000041	-	5	1533	5307	3823	5311	3823	1533	-	0	1353.7	36.3	-
Archaeal_16S_rRNA	RF01959	K_rhaeticus_N_B2_000000000041	-	7	1477	5300	3825	5305	3825	1477	-	8.6e-232	764.4	35.7	-
Bacterial_23S_rRNA	-	K_rhaeticus_TNT2_B2_000000000052	-	11	2900	3056	331	3066	325	2906	-	0	2384.1	63.6
Bacterial_16S_rRNA	RF00177	K_rhaeticus_TNT2_B2_000000000052	-	5	1533	5307	3823	5311	3823	1533	-	0	1353.7	36.3
Archaeal_16S_rRNA	RF01959	K_rhaeticus_TNT2_B2_000000000052	-	7	1477	5300	3825	5305	3825	1477	-	8.6e-232	764.4	35.7
```

**Comment by meren** [comment]
Quick question. Does the same thing happen when you run the workflow with a single thread?

And are you running the control run outside of the workflow with multiple threads?

**Comment by efogarty11** [comment]
> Does the same thing happen when you run the workflow with a single thread?

Yes, it gives me the same error (I just checked 🙂)

> And are you running the control run outside of the workflow with multiple threads?

what do you mean by control run? Where I ran hmms manually or where I ran the workflow without clusterize? When I ran hmms manually (outside the workflow) it works fine with 5 threads, and when I ran the workflow without clusterize it still failed (although come to think if it I may have forgotten to change the threads in the config file when I ran it without clusterize, so it was maybe still running with 8 threads? I can go back and try this again if this is what you meant).

**Comment by meren** [comment]
Thanks! I think you've answered my question, although what might be causing this has not become any clearer in my mind :)

**Comment by meren** [comment]
What happened to this, @efogarty11?

**Comment by ekiefl** [comment]
@meren I am going to take a look at this before v7. Are you planning on releasing a v7 alpha? If so, when?

**Comment by meren** [comment]
Thanks, Evan. I'm changing the HMM models at the moment. It is better if you tested after that I think.

> Are you planning on releasing a v7 alpha? If so, when?

IDK yet :/

**Comment by efogarty11** [comment]
@meren sorry I just saw this. If there's anything I can test/help with please let me know!

**Comment by meren** [comment]
This is discussed and resolved in #1564.
